R/pkg/R/window.R

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# window.R - Utility functions for defining window in DataFrames
+
+#' Window.partitionBy
+#'
+#' Creates a WindowSpec with the partitioning defined.
+#'
+#' @rdname Window.partitionBy
+#' @name Window.partitionBy
+#' @export
+#' @examples
+#' \dontrun{
+#'   Window.partitionBy("col1", "col2")
+#'   Window.partitionBy(df$col1, df$col2)
+#' }
+setMethod("Window.partitionBy",
+          signature(col = "character"),
+          function(col, ...) {
+            windowSpec(
+              callJStatic("org.apache.spark.sql.expressions.Window",
+                          "partitionBy",
+                          col,
+                          list(...)))
+          })
+
+#' @rdname Window.partitionBy
+#' @name Window.partitionBy
+#' @export
+setMethod("Window.partitionBy",
```

Comment:
We already have S4 functions for those in org.apache.spark.sql.functions in Scala which don't have a class. I could implement these 2 as S3 or normal functions. But I think implementing them as S4 functions are consistent and having the benefit of checking parameter types.